contrib/opencensus-ext-azure/opencensus/ext/azure/metric_exporter/__init__.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2019, OpenCensus Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import threading
+
+from opencensus.ext.azure.common import Options
+from opencensus.metrics import transport
+from opencensus.stats import stats
+
+__all__ = ['MetricsExporter']
+
+class MetricsExporter(object):
```

Comment:
Regarding the name, I wish to discuss about (we might want to do a better job in OpenTelemetry):
1. Do we want to drive for some level of consistency or not at all (and leave this for individual vendor's choice of flavor)?
2. When to use plural form versus not?
3. Should the class name be specific or generic? (e.g. do we want AzureExporter, AzureMonitorExporter, AzureMetricExporter or AzureMetricsExporter, my personal preference would be something we did for Azure trace exporter and stackdriver trace exporter)
4. Should the class name be self contained or we should leverage package name? (e.g. `ext.azure.trace_exporter.AzureExporter`, `ext.azure.TraceExporter` or `ext.azure.AzureTraceExporter`).

https://github.com/census-instrumentation/opencensus-python/blob/master/contrib/opencensus-ext-azure/opencensus/ext/azure/trace_exporter/__init__.py#L34

https://github.com/census-instrumentation/opencensus-python/blob/master/contrib/opencensus-ext-stackdriver/opencensus/ext/stackdriver/trace_exporter/__init__.py#L164

Reply:
Yes I was thinking about this as well. I think it is good to be consistent across the entire project and set some standards.

In my opinion:
1. We should use singular form when referring to the "task-doer" such as MetricExporter, TraceExporter, etc. We should use the plural when we are referring to the category in a collective context (this will export metrics..., the traces are sent through..., etc) because the logic involves one or possibly more of these.

2. In regards to the class name, I think we should be as specific as possible. When going through the codebase, it is easier to identify what module I am looking at without having to look at the package I am under (problem that I am having with init.py). Also, if we ever take code snippets, readers can identify what class is being looked at without having to see the file path.

3. In regards to leveraging the package name, separating the files based off levels is my preference. So ext.azure.trace_exporter.AzureTraceExporter would be ideal.

Reply:
@c24t what's your opinion?

Seems in OpenTelemetry Java we've already started to diverge (metrics and tags are plural form, while trace is not):

https://github.com/open-telemetry/opentelemetry-java/tree/master/api/src/main/java/io/opentelemetry

Reply:
We have just discussed the exact same issue recently, in https://github.com/open-telemetry/opentelemetry-java/issues/259#issuecomment-496999460. The conclusion is:

> Let's stick with what we have, but pluralize resources.
trace
metrics
tags
resources

Reply:
What about log? Using `logs` seems to be against language convention.

Reply:
We haven't talked about it but personally I'd suggest to go with `log` for now. Agree `logs` doesn't seem to be a great name here.

Reply:
I will make the changes based off these decisions for metrics.